.github/workflows/build.yml

```diff
@@ -9,9 +9,10 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        os: [ubuntu-latest, windows-latest]
```

Comment:
Should we also run on macOS, since it's also a popular OS for this?

Reply:
The only platform specific steps we do in the code base is the file paths and handling of slashes `/` and `\` in them. We could add macos-latest just to be sure off course. Just seems to heavy to me. I originally made this PR after the issue claimed that there were 8 test failures on Windows. Maybe if nothing else, running the pipeline on all platforms can show people who have build issues that it does indeed work on their platform? (and that it is probably their machines that have some sort of issue?)

Reply:
Yeah, that's exactly what I'm thinking. If we were to run it on that OS too, we could build a lot more confidence in saying that there aren't any issues spin each of the major systems. Happy to do it in another PR though if you like?

Reply:
Added in in my latest commit 🙂 Will give it a day before merge just in case @fwcd has any views. The argument on building confidence should be pretty clear though. If we build for all major OSs, then we can link the latest main build in build-related issues people might have 🙂

Reply:
Sounds reasonable. 👍 I wouldn't expect a lot of differences between macOS/Linux either, but who knows, building (and testing) on all major OSes is probably not a bad idea in any case. Even though Java tends to abstract pretty well over the platform, small things like path separators can quickly go unnoticed in such cases.

Though to be fair, these things are generally also dependent on the client and testing the server in isolation might not be sufficient to catch all OS-specific regressions (e.g. our test cases might use POSIX paths on Windows too while other clients send backslashes).